### PR TITLE
refactor Dartino SOD launch to use new SOD debug daemon

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew

### CODE REVIEW

File: lib/dartino/device/device.dart [hunk 1]
```diff
@@ -51,5 +53,71 @@ abstract class Device {
 
   /// Launch the specified application on the device and return `true`.
   /// If there is a problem, notify the user and return `false`.
-  Future<bool> launchSOD(SodRepo sdk, DartinoLaunch launch);
+  Future<bool> launchSOD(SodRepo sdk, DartinoLaunch launch) async {
+    //TODO(danrubel) add windows and mac support and move this into cmdline util
+    if (isWindows && isMac) {
```
Comment (devoncarew) [inline lib/dartino/device/device.dart]: `||`?

File: lib/dartino/device/device.dart [hunk 2]
```diff
@@ -51,5 +53,71 @@ abstract class Device {
...
+    String destPath;
+    destPath = '/spifs/${fs.basename(binPath)}';
+    //TODO(danrubel) remove next line once files can be removed or overwritten
+    // destPath = '/spifs/d${new DateTime.now().millisecondsSinceEpoch}.snap';
+    // Start background daemon if not already running
+    LkShell shell = await sdk.startDebugDaemon(launch);
+    if (shell == null) return false;
+    // Deploy to device
+    shell.write('\n');
+    await new Future.delayed(new Duration(milliseconds: 250));
```
Comment (devoncarew) [inline lib/dartino/device/device.dart]: Is there a way to know when it's safe to proceed, or is this delay the most practical method?

Comment (devoncarew) [inline lib/dartino/device/device.dart]: You may want to add a short utility to return a delayed Future `await _delay(250);`.

**Comment by devoncarew** [comment]
Are you using the console UI for your launches? Some of the output from LkShell may be worth displaying there, to debug launch failures.

File: lib/dartino/device/device.dart (hunk 1 above)
Comment (danrubel) [inline lib/dartino/device/device.dart]: Good catch. Fixed.

File: lib/dartino/device/device.dart (hunk 2 above)
Comment (danrubel) [inline lib/dartino/device/device.dart]: This delay is the most expedient way for now. This code will be ripped out once the underlying debug daemon is fixed to handle resetting the device, removing the file, etc.

**Comment by danrubel** [comment]
All of the LkShell stdout and stderr is [piped to the launch console](https://github.com/dart-atom/dartlang/pull/984/files#diff-dc2f33cbf8defff3e07b4c260d6869deR190).

**Comment by devoncarew** [comment]
> All of the LkShell stdout and stderr is piped to the launch console.

Ah, missed that.

lgtm!

**Comment by danrubel** [comment]
Merged https://github.com/dart-atom/dartlang/commit/3af70ec57ddf991acdde03d6384d98482fce751b

**Pull request closed without merging**